### PR TITLE
Update link, dropdown-menu-nav, and popover to use a11y-label - Breaking

### PR DESCRIPTION
**Pull request by sirrah-tam**
**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [x] Yes
- [ ] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [ ] Code coverage maximal?
- [x] Changeset added?
- [ ] Component status page up to date?

**What does this change address?**
In an effort to use consistent naming conventions for ARIA attributes, this change follows the updates for the Pharos Button to use `a11y-`. 

This impacts the following components:

- pharos-dropdown-menu-nav
- pharos-link
- pharos-popover

This also addresses #674 as the breaking change.

**How does this change work?**
This adds the `a11y-label` attribute and uses it to apply the affected components with that `aria-label` attribute. Consumers will need to update their instances of `label` for these components or will run into errors passing accessibility checks.

### CODE REVIEW

File: packages/pharos/src/components/dropdown-menu-nav/pharos-dropdown-menu-nav.ts [hunk 1]
```diff
@@ -19,11 +19,11 @@ import FocusMixin from '../../utils/mixins/focus';
  */
 export class PharosDropdownMenuNav extends FocusMixin(PharosElement) {
   /**
-   * Indicates the aria label to apply to the nav.
-   * @attr label
+   * Indicates the aria label to apply to the button.
```
Comment (daneah) [inline packages/pharos/src/components/dropdown-menu-nav/pharos-dropdown-menu-nav.ts:22]: Will a consumer using PharosDropdownMenuNav know what `button` is being referred to here? I suspect not, personally. Might do with a bit more elaboration.

File: packages/pharos/src/components/link/pharos-link.ts [hunk 2]
```diff
@@ -43,11 +43,11 @@ export class PharosLink extends FocusMixin(AnchorElement) {
   public isOnBackground = false;
 
   /**
-   * Indicates the aria label to apply to the link.
-   * @attr label
+   * Indicates the aria label to apply to the button.
```
Comment (daneah) [inline packages/pharos/src/components/link/pharos-link.ts:46]: Same here—possibly even incorrect, in some cases?

File: packages/pharos/src/components/popover/pharos-popover.ts [hunk 3]
```diff
@@ -38,11 +38,11 @@ export class PharosPopover extends ScopedRegistryMixin(FocusMixin(OverlayElement
   public isOnBackground = false;
 
   /**
-   * Indicates the aria label to apply to the dialog.
-   * @attr label
+   * Indicates the aria label to apply to the button.
```
Comment (daneah) [inline packages/pharos/src/components/popover/pharos-popover.ts:41]: I'm not sure PharosPopover ever _renders_ a button. Looks like the label goes on a `div` in this case.

File: packages/pharos/src/components/dropdown-menu-nav/pharos-dropdown-menu-nav.ts (hunk 1 above)
Comment (sirrah-tam) [reply packages/pharos/src/components/dropdown-menu-nav/pharos-dropdown-menu-nav.ts:22]: Oof, you are correct, a lil bit of copy/pasta action happening here. Will update accordingly!

**Review by daneah — APPROVED** [review]

**Pull request merged**